src/components/internal/FormFields/FileInput/components/Dropzone.tsx

```diff
@@ -115,7 +120,13 @@ export const Dropzone = fixedForwardRef<DropzoneProps, HTMLInputElement>((props,
                     if (!allowedFileTypes.includes(file.type)) {
                         throw validationErrors.DISALLOWED_FILE_TYPE;
                     }
-                    if (file.size > maxFileSize) {
+
+                    // Determine the current max file size for the file type
+                    const currentMaxFileSize = typeof maxFileSize === 'function' ? maxFileSize(file.type) || DEFAULT_MAX_FILE_SIZE : maxFileSize;
```

Comment:
You can use the `isFunction()` util instead.
```suggestion
                    const currentMaxFileSize = isFunction(maxFileSize) ? maxFileSize(file.type) ?? DEFAULT_MAX_FILE_SIZE : maxFileSize;
```

Reply:
✅